controllers/apply/awards-for-all/fields.js

```diff
@@ -2093,76 +2139,75 @@ module.exports = function fieldsFor({ locale, data = {} }) {
                 ];
             }
         },
-        mainContactName: nameField(
-            {
-                name: 'mainContactName',
-                label: localise({ en: 'Full name of main contact', cy: '' }),
-                explanation: localise({
-                    en: 'This person has to live in the UK.',
-                    cy: ''
-                }),
-                get warnings() {
-                    let result = [];
+        mainContactName: nameField({
+            name: 'mainContactName',
+            label: localise({ en: 'Full name of main contact', cy: '' }),
+            explanation: localise({
+                en: 'This person has to live in the UK.',
+                cy: ''
+            }),
+            get warnings() {
+                let result = [];
 
-                    const seniorSurname = get('seniorContactName.lastName')(
-                        data
-                    );
+                const seniorSurname = get('seniorContactName.lastName')(data);
 
-                    const lastNamesMatch =
-                        seniorSurname &&
-                        seniorSurname === get('mainContactName.lastName')(data);
+                const lastNamesMatch =
+                    seniorSurname &&
+                    seniorSurname === get('mainContactName.lastName')(data);
 
-                    if (lastNamesMatch) {
-                        result.push(
-                            localise({
-                                en: `<span class="js-form-warning-surname">We've noticed that your main and senior contact
+                if (lastNamesMatch) {
+                    result.push(
+                        localise({
+                            en: `<span class="js-form-warning-surname">We've noticed that your main and senior contact
                                      have the same surname. Remember we can't fund projects
                                      where the two contacts are married or related by blood.</span>`,
-                                cy: ``
-                            })
-                        );
-                    }
+                            cy: ``
+                        })
+                    );
+                }
 
-                    return result;
-                },
-                schema: Joi.fullName()
-                    .mainContact()
-                    .required()
+                return result;
             },
-            [
+            schema: Joi.object({
+                firstName: Joi.string().required(),
+                lastName: Joi.string().required()
+            })
+                .compare(Joi.ref('seniorContactName'))
```

Comment:
We have a new low level `compare-object` extension which allows us to pass a ref to do a deep `isEqual` check